assets/src/common/components/pre-publish-panel.js

```diff
@@ -1,73 +1,26 @@
-/**
- * External dependencies
- */
-import PropTypes from 'prop-types';
-
 /**
  * WordPress dependencies
  */
-import { Notice } from '@wordpress/components';
+import { PostFeaturedImage } from '@wordpress/editor';
 import { PluginPrePublishPanel } from '@wordpress/edit-post';
-import { withSelect } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 
 /**
- * Internal dependencies
- */
-import { validateFeaturedImage } from '../helpers';
-
-/**
- * Conditionally adds a notice to the pre-publish panel for the featured image.
+ * Adds a notice to the pre-publish panel for the featured image.
  *
- * @param {Object}  props               Component props.
- * @param {Object}  props.featuredMedia Media object.
- * @param {Array}   props.dimensions    Required image dimensions.
- * @param {boolean} props.required      Whether selecting a featured image is required.
  * @return {Function} Either a plain pre-publish panel, or the panel with a featured image notice.
```

Comment:
The `@return` description will have to be updated.

```suggestion
 * @return {Function} A pre-publish panel containing the featured image selection component.
```